### ISSUE TITLE
Displaying RSS feed correctly after clicking on a RSS link in a Web page

### ISSUE THREAD

**Issue by secou** [Reeywhaar/want-my-rss#10]
Hi, 

It would be deeply helpful if the extension could display correctly a feed when ones click on a hyperlink in a web page pointing to a RSS feed, like, say, the ones we find in the [New York Times RSS page](https://archive.nytimes.com/www.nytimes.com/services/xml/rss/index.html) !

Serge

**Comment by Reeywhaar** [comment]
You mean links like this?
![screen shot 2018-12-22 at 23 23 36](https://user-images.githubusercontent.com/884649/50378345-18c95780-0641-11e9-8c03-7740e192fd1d.png)

Well, the problem is there is now way to distinguish between whether link points on feed, or is it just some content link.
![screen shot 2018-12-22 at 23 30 40](https://user-images.githubusercontent.com/884649/50378365-ad33ba00-0641-11e9-9a92-a42c23794383.png)

But I think of a way: I could intercept web requests, look at their content-type and look at header, whether it conforms rss or atom spec. This way it will be even more like this was in old Firefox, because there will be no `moz-extension://{ext-id}/show.html?url={feed-url}` in url bar, there will be just feed url. There is a problem that many websites output feeds without proper content type, it could be `text/plain` or something.

Right now there is a workaround, you can right click on a link, copy it's address and type `ext+rss:` in url bar and paste the link.

**Comment by secou** [comment]
OK I understand the problem!
In Chrome, a subscribtion extension does that pretty well.
https://chrome.google.com/webstore/detail/rss-subscription-extensio/nlbjncdgjeocebhnmkbbbdekmmmcbfjd
But perhaps is it difficult to do the same thing now in Firefox.
Thanks again for your work!

**Comment by jscher2000** [comment]
Another possible workaround: add a command to the right-click context menu to load a link in the RSS viewer (show.html). As an example of how it might work (not tested):

````
/**** Create context menu item ****/
let linkcontext = browser.menus.create({
  id: "wantmyrss_content_link",
  title: "Open in Want my RSS",
  contexts: ["link"],
  icons: {
	"icons/icon-color.svg"
  }
});
/**** Handle context menu item click ****/
browser.menus.onClicked.addListener((menuInfo, currTab) => {
	switch (menuInfo.menuItemId) {
		case 'wantmyrss_content_link':
			// Probably only works with HTTP and HTTPS - link check TODO
			var RUrl = browser.extension.getURL('show.html') + '?url=' + encodeURIComponent(menuInfo.linkUrl);
			browser.tabs.create({
				url: RUrl,
				active: true
			}).then((newTab) => {
				/* No action */;
			}).catch((err) => {
				console.log('Error in tabs.create with "' + RUrl + '": ' + err.message);
			});
		default:
			// Error?
	}
});
````

**Comment by Reeywhaar** [comment]
@jscher2000 thank you for the concrete example! I'm still thinking about solution, expecially @secou gave an example of extension that works.

**Comment by jscher2000** [comment]
The interception works for me. I tested with:

(A) The links in the right column of my blog under "Site Actions": https://www.jeffersonscher.com/ttw/ 

These send Content-Type: application/rss+xml; charset=UTF-8

_Same Content-Type sent by:_ https://www.nasa.gov/rss/dyn/lg_image_of_the_day.rss

(B) The "RSS LINK" button on this podcast: https://www.npr.org/podcasts/510313/how-i-built-this

This sends Content-Type: text/xml;charset=UTF-8

So that's very promising. I don't have examples of other possible Content-Type's to check.

**Comment by jscher2000** [comment]
On the question of performance as well as false positives, since you only intercept main_frame requests, I think the risks are low.

**Comment by Reeywhaar** [comment]
Yeah, I was afraid at first, but it seems it doens't hurt experience much if any. My concern was that cumulative request blocking of multiple extensions would kill speed, so didn't want to add to this cauldron.

**Comment by jscher2000** [comment]
I think with this change, you can reply to the review from a few days ago: 

https://addons.mozilla.org/firefox/addon/want-my-rss/reviews/1257481/

**Comment by Reeywhaar** [comment]
Thank you for review! Replied to user.
